### PR TITLE
chore: update nightly-testing-2024-12-23

### PR DESCRIPTION
**Pull request by luisacicolini**
WIP. integrating changes from #907 too - cc @tobiasgrosser: sorry i did not see #907 existed already when i started working on this

### CODE REVIEW

**Comment by tobiasgrosser** [comment]
All good. Thank you for looking into this. When updating mathlib, it makes sense to update incrementally to individual releases as this will (a) make `git bisect` easier and (b) allows us to fix one issue at a time. I will add some comments what I am currently struggling with in https://github.com/opencompl/lean-mlir/pull/907.

**Comment by luisacicolini** [comment]
sure, then I'm happy to help with #907 and then come back to this one!

**Comment by luisacicolini** [comment]
There is just one issue with `rfl` in `FormulaToAuto/Language.mem_setOf_eq`. Working on it.

**Comment by tobiasgrosser** [comment]
I am surprised this is an issue at all. Afaiu I lean itself nothing changed. So I am unsure how this proof must now be different. Did sth in Mathlib change?

**Comment by luisacicolini** [comment]
There were some changes made to `Language.lean` in Mathlib, indeed

**Comment by luisacicolini** [comment]
but I don't see how that would impact the definition (https://github.com/leanprover-community/mathlib4/compare/nightly-testing-2024-12-21...nightly-testing-2024-12-23?diff=split&w=#diff-eeee6783198a232a3a9566d67c87f0073ac7b01bf79f0db8cdd5580d8497bbe8)

**Comment by bollu** [comment]
@luisacicolini It was renamed to `Language.instMembershipList`. Oh the joy of auto-implicits, where `instMembershipListLanguage` was now a variable, which lean didn't know how to make head or tails of :)

**Comment by bollu** [comment]
```
subprocess.TimeoutExpired: Command 'lake lean /home/tobiascgrosser/actions-runner/_work/lean-mlir/lean-mlir/SSA/Projects/InstCombine/tests/proofs/gsubhfromhsub_proof.lean' timed out after 1800 seconds
```

File: SSA/Experimental/Bits/AutoStructs/FormulaToAuto.lean [hunk 1]
```diff
@@ -19,9 +19,9 @@ import SSA.Experimental.Bits.AutoStructs.NFA'
 open AutoStructs
 open Mathlib
 
+set_option pp.explicit true in
 @[simp] theorem Language.mem_setOf_eq {x : List α} {p : List α → Prop} :
```
Comment (tobiasgrosser) [inline SSA/Experimental/Bits/AutoStructs/FormulaToAuto.lean]:
```suggestion
@[simp] theorem Language.mem_setOf_eq {x : List α} {p : List α → Prop} :
```

**Comment by tobiasgrosser** [comment]
> ```
> subprocess.TimeoutExpired: Command 'lake lean /home/tobiascgrosser/actions-runner/_work/lean-mlir/lean-mlir/SSA/Projects/InstCombine/tests/proofs/gsubhfromhsub_proof.lean' timed out after 1800 seconds
> ```

This is expected. :-) I should disable the error. (These timeouts are orthogonal)

**Pull request merged**